train_ScanNet_DDP_WarmUP.py

```diff
@@ -503,6 +511,11 @@ def train(train_loader, model, criterion, optimizer, epoch, scheduler):
 #                                'mAcc_train_batch': np.mean(intersection / (target + 1e-10)),
 #                                'allAcc_train_batch': accuracy,
 #                                'lr_train_batch': lr}
+
+        del inv_self, inv_forward, inv_propagate
+        del pred, loss, output, intersection, union, target
+        torch.cuda.empty_cache()
```

Comment:
Are these lines necessary to ensure that memory doesn't blow up during training? If so, we need a work around this as we can't expect users to keep writing this every time they perform a forward+backward pass.